src/vs/platform/native/electron-main/nativeHostMainService.ts

```diff
@@ -488,7 +490,17 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 
 	async openExternal(windowId: number | undefined, url: string): Promise<boolean> {
 		this.environmentMainService.unsetSnapExportedVariables();
-		shell.openExternal(url);
+		const defaultBrowser = this.configurationService.getValue<string>('workbench.defaultBrowser');
```

Comment:
This method is being called for anything to open, not just URLs. The URL can be a local file protocol too, so I would restrict this to `http` and `https` only:

```ts
if (matchesScheme(url, Schemas.http) || matchesScheme(url, Schemas.https)) {
```